code/modules/clothing/suits/armor.dm

```diff
@@ -36,9 +36,43 @@
 
 /obj/item/clothing/suit/armor/vest/security
 	name = "security armor"
-	desc = "An armored vest that protects against some damage. This one has Nanotrasen corporate badge."
-	icon_state = "armorsec"
+	desc = "An armored vest that protects against some damage. This one has a clip for a holobadge."
+	icon_state = "armor"
 	item_state = "armor"
+	var/obj/item/clothing/accessory/holobadge/attached_badge
+
+/obj/item/clothing/suit/armor/vest/security/attackby(obj/item/W as obj, mob/user as mob, params)

```

Comment:
I realized I'm an idiot and that this is, in fact, a suit and not uniform. Yeah, this is the best way to handle it.